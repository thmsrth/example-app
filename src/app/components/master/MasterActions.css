.MasterActions {
  margin: var(--section-margin) 0 0;
}

.MasterActions__icon {
  margin: 0 var(--section-margin);
}

.MasterActions__container {
  padding: var(--section-padding) var(--section-padding) 0;
  display: inline-flex;
}

.MasterActions__button {
  background-color: var(--color-master-action);
  margin-right: var(--section-margin);
}

.MasterActions__button .MasterActions__button__icon,
.MasterActions__button .MasterActions__button__text {
  color: var(--color-black);
}

.MasterActions__button--disabled {
  background-color: var(--button-disabled-background-color);
}

.MasterActions__button--disabled .MasterActions__button__icon,
.MasterActions__button--disabled .MasterActions__button__text {
  color: var(--button-disabled-color);
}
