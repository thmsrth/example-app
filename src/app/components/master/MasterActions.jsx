import React from 'react';

import { MAIN_PATH } from '../../../../config/constants';

import { historyPush } from '../../../global/utils/UrlHelpers';
import Button from '../../../global/components/Button';

import I18n from '../../utils/I18n';
import SecurityUtil from '../../../global/security/utils/SecurityUtil';

import './MasterActions.css';

const MasterActions = () => (
  <div className="MasterActions">
    <div className="MasterActions__container">
      <Button
        className="MasterActions__button"
        text={I18n.t('en.actions.create')}
        iconName="plus"
        onClick={() => historyPush(`${MAIN_PATH}/create`)}
        disabled={!SecurityUtil.isGranted('EXAMPLE_CREATE')}
      />
    </div>
  </div>
);

export default MasterActions;
