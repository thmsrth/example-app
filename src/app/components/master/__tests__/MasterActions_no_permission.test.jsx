import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import MasterActions from '../MasterActions';

global.permissions = {};

const props = { myCreateAction: () => {} };

describe('MasterActions', () => {
  it('should render the component correctly with disabled create button', () => {
    const tree = shallow(<MasterActions {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
