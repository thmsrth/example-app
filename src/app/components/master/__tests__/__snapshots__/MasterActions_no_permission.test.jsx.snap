// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterActions should render the component correctly with disabled create button 1`] = `
<div
  className="MasterActions"
>
  <div
    className="MasterActions__container"
  >
    <ButtonWrapper
      className="MasterActions__button"
      disabled={true}
      iconName="plus"
      onClick={[Function]}
      params={Object {}}
      text="Add"
    />
  </div>
</div>
`;
